### PR TITLE
Add JoinOperation examples

### LINKED ISSUE #16237: Examples?
Like other LINQ articles
Query expression syntax examples for `Join`, `GroupJoin` would be nice

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4cc59b33-c264-8ccd-a0e3-e916d1ce676f
* Version Independent ID: 8f71c9ce-d29b-da24-8262-e1ab82fe4cc9
* Content: [Join Operations (C#)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/join-operations)
* Content Source: [docs/csharp/programming-guide/concepts/linq/join-operations.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/concepts/linq/join-operations.md)
* Product: **dotnet-csharp**
* Technology: **csharp-linq**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by SuperNova911**
Link CsLINQ join operation examples with dotnet/samples#1876

Fixes #16237

### CODE REVIEW

File: docs/csharp/programming-guide/concepts/linq/join-operations.md [hunk 1]
```diff
@@ -21,6 +21,20 @@ A *join* of two data sources is the association of objects in one data source wi
 |Join|Joins two sequences based on key selector functions and extracts pairs of values.|`join … in … on … equals …`|<xref:System.Linq.Enumerable.Join%2A?displayProperty=nameWithType><br /><br /> <xref:System.Linq.Queryable.Join%2A?displayProperty=nameWithType>|  
 |GroupJoin|Joins two sequences based on key selector functions and groups the resulting matches for each element.|`join … in … on … equals … into …`|<xref:System.Linq.Enumerable.GroupJoin%2A?displayProperty=nameWithType><br /><br /> <xref:System.Linq.Queryable.GroupJoin%2A?displayProperty=nameWithType>|  
   
+## Query Expression Syntax Examples  
+  
+### Join  
+  
+The following example uses the `join … in … on … equals …` clause to join two sequences based on specific value
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/concepts/linq/join-operations.md]:
```suggestion
The following example uses the `join … in … on … equals …` clause to join two sequences based on specific value:
```

File: docs/csharp/programming-guide/concepts/linq/join-operations.md [hunk 2]
```diff
@@ -21,6 +21,20 @@ A *join* of two data sources is the association of objects in one data source wi
...
+  
+[!code-csharp[Join](~/samples/snippets/csharp/VS_Snippets_VBCSharp/CsLINQJoin/CS/JoinOperation.cs#Join)]  
+
+### GroupJoin  
+
+The following example uses the `join … in … on … equals … into …` clause to join two sequences based on specific value and groups the resulting matches for each element
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/concepts/linq/join-operations.md]:
```suggestion
The following example uses the `join … in … on … equals … into …` clause to join two sequences based on specific value and groups the resulting matches for each element:
```

**Review by Youssef1313 — CHANGES_REQUESTED** [review]

File: docs/csharp/programming-guide/concepts/linq/join-operations.md [hunk 3]
```diff
@@ -21,6 +21,20 @@ A *join* of two data sources is the association of objects in one data source wi
 |Join|Joins two sequences based on key selector functions and extracts pairs of values.|`join … in … on … equals …`|<xref:System.Linq.Enumerable.Join%2A?displayProperty=nameWithType><br /><br /> <xref:System.Linq.Queryable.Join%2A?displayProperty=nameWithType>|  
 |GroupJoin|Joins two sequences based on key selector functions and groups the resulting matches for each element.|`join … in … on … equals … into …`|<xref:System.Linq.Enumerable.GroupJoin%2A?displayProperty=nameWithType><br /><br /> <xref:System.Linq.Queryable.GroupJoin%2A?displayProperty=nameWithType>|  
   
+## Query Expression Syntax Examples
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/concepts/linq/join-operations.md]:
```suggestion
## Query expression syntax examples
```

File: docs/csharp/programming-guide/concepts/linq/join-operations.md [hunk 4]
```diff
@@ -21,6 +21,20 @@ A *join* of two data sources is the association of objects in one data source wi
 |Join|Joins two sequences based on key selector functions and extracts pairs of values.|`join … in … on … equals …`|<xref:System.Linq.Enumerable.Join%2A?displayProperty=nameWithType><br /><br /> <xref:System.Linq.Queryable.Join%2A?displayProperty=nameWithType>|  
 |GroupJoin|Joins two sequences based on key selector functions and groups the resulting matches for each element.|`join … in … on … equals … into …`|<xref:System.Linq.Enumerable.GroupJoin%2A?displayProperty=nameWithType><br /><br /> <xref:System.Linq.Queryable.GroupJoin%2A?displayProperty=nameWithType>|  
   
+## Query Expression Syntax Examples  
+  
+### Join
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/concepts/linq/join-operations.md]: Should localization be avoided here ?

File: docs/csharp/programming-guide/concepts/linq/join-operations.md [hunk 5]
```diff
@@ -21,6 +21,20 @@ A *join* of two data sources is the association of objects in one data source wi
...
   
+## Query Expression Syntax Examples  
+  
+### Join  
+  
+The following example uses the `join … in … on … equals …` clause to join two sequences based on specific value:
+  
+[!code-csharp[Join](~/samples/snippets/csharp/VS_Snippets_VBCSharp/CsLINQJoin/CS/JoinOperation.cs#Join)]  
+
+### GroupJoin
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/concepts/linq/join-operations.md]: Same here ?

File: docs/csharp/programming-guide/concepts/linq/join-operations.md (hunk 4 above)
Comment (BillWagner) [reply docs/csharp/programming-guide/concepts/linq/join-operations.md]: Yes, we should avoid localization.  @JasonCard What's the best way to ensure that in an H3 heading?

**Comment by BillWagner** [comment]
This is ready once we get an answer on the localization question, and the related samples PR is merged.

File: docs/csharp/programming-guide/concepts/linq/join-operations.md (hunk 4 above)
Comment (JasonCard) [reply docs/csharp/programming-guide/concepts/linq/join-operations.md]: @BillWagner You can use the no-loc metadata mentioned [here](https://review.docs.microsoft.com/en-us/help/contribute/markdown-reference?branch=master#non-localized-strings), for example:
---
title: Title of the Document
author: author-name
description: Description for the document
no-loc: [Join, GroupJoin]
---

Comment (BillWagner) [reply docs/csharp/programming-guide/concepts/linq/join-operations.md]: @SuperNova911 Can you add the line referenced above in Jason's comment? Then, this is ready to :shipit:

File: docs/csharp/programming-guide/concepts/linq/join-operations.md (hunk 5 above)
Comment (JasonCard) [reply docs/csharp/programming-guide/concepts/linq/join-operations.md]: Adding this line below within the YAML header will lock all instances of both "Join" and "GroupJoin":
```
no-loc: [Join, GroupJoin]
```

**Review by JasonCard — APPROVED** [review]
no-loc metadata looks good.

**Review by BillWagner — APPROVED** [review]
This looks great now @SuperNova911 

I'll :shipit: now.

**Comment by tdykstra** [comment]
@BillWagner something went wrong with this or the samples PR -- I'm getting snippet not found errors in the build and the snippets aren't showing in production.  I took a look and didn't see anything immediately obvious.  Will look again later.

**Comment by BillWagner** [comment]
@tdykstra I restored the missing file in dotnet/samples#1936 I can't figure out what happened though.

**Pull request merged**